sdk/storage/inc/datalake/path_client.hpp

```diff
@@ -207,18 +226,71 @@ namespace Azure { namespace Storage { namespace DataLake {
     /**
      * @brief Sets the properties of a resource the path points to.
      * @param options Optional parameters to set the properties to the resource the path points to.
-     * @return PathUpdateResponse
+     * @return SetPropertiesResult
      */
-    PathUpdateResponse SetProperties(
+    SetPropertiesResult SetProperties(
         const SetPathPropertiesOptions& options = SetPathPropertiesOptions()) const;
 
+    /**
+     * @brief Get Properties returns all system and user defined properties for a path. Get Status
+     *        returns all system defined properties for a path. Get Access Control List returns the
+     *        access control list for a path.
+     * @param options Optional parameters to get the properties from the resource the path points
+     *                to.
+     * @return GetPathPropertiesResponse
+     */
+    GetPathPropertiesResponse GetProperties(
+        const PathGetPropertiesOptions& options = PathGetPropertiesOptions()) const;
+
+    /**
+     * @brief Sets the metadata of a resource the path points to.
+     * @param options Optional parameters to set the metadata to the resource the path points to.
+     * @return SetMetadataResult
+     */
+    SetMetadataResult SetMetadata(
+        const std::map<std::string, std::string> metadata,
+        const SetPathMetadataOptions& options = SetPathMetadataOptions()) const;
+
+    /**
+     * @brief Get the metadata for a path.
+     * @param options Optional parameters to get the metadata from the resource the path points to.
+     * @return std::map<std::string, std::string>
+     */
+    std::map<std::string, std::string> GetMetadata(
```

Comment:
We don't need this API

Reply:
It is just much nicer to have a `GetMetadata` returning directly the `metadata`. Calling `GetProperties` is not intuitive enough for retrieving metadata.

Reply:
https://github.com/Azure/azure-sdk-for-cpp/pull/257#discussion_r452608647